packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1789,6 +1789,14 @@ class ArtifactUpdater {
       } finally {
         status.stop();
       }
+      /// Unzipping multiple file into a directory will not remove old files
+      /// from previous versions that are not present in the new bundle.
+      final Directory destination = location.childDirectory(
+        tempFile.fileSystem.path.basenameWithoutExtension(tempFile.path)
+      );
+      if (destination.existsSync()) {
+        destination.deleteSync(recursive: true);
```

Comment:
What should/will happen if this throws an exception?

Reply:
hmm, I'm not really sure. I don't know if - say there was a missing permission, would the unzip fail anyway? I can rethrow a tool exit to start?

Reply:
actually, the ones we care about should go to error handling io, for the others we can let it crash.

Reply:
It looks like `deleteSync` is already covered by the ErrorHandlingFileSystem. Additionally it looks like this:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/error_handling_io.dart#L74

covers some additional cases that might be useful here?

Reply:
Done